content/purgatorio/collect-and-prepare-data/data-collection.md

```diff
@@ -0,0 +1,209 @@
+
+[Back to the index](#index)
+
+### Data Generation
+
+Data generation is also a good MVP step/process, and also could be incorporated as one of your steps during the first few iterations - if you discover or know from observation that the data to hand is not sufficient or correct or even not usable for the end-purpose (see [Research or business questions](#Research-or-business-questions)).
+
+It is often considered when a domain or field has limited data or previous experiments or exercises have resulted in a small dataset but the dataset may be some what representative of the domain or problem that is being solved - then Data Generation using existing could be take resort to. This of course has it's pros and cons - you soon get a lot of data but then is the data biased and in what way? Is it balanced or not, if not what to do - all these kinds of questions come to the surface.
+
+This topic may need a section on it's own but there are many resources and facilities propping up these days that do exactly this, see this resource on [Data Generation](https://github.com/neomatrix369/awesome-ai-ml-dl/blob/master/data/data-generation.md#data-generation).
+
+[Back to the index](#index)
+
+### Visualisation
+
+"A picture speaks a thousand words" or "A picture is worth a thousand words" or other variants - they all mean the same thing!
+
+There is a misconception that Data Visualisation takes place towards the end or during some other process in the end-to-end process. But the fact is, whenever you have data, no matter what state it is in, we can partially (if not fully) visualise it. And gain an insight into what it is representing and what state it is in. It is an art and not just a science to be able to do this, the science part of drawing pictures from it is more or less solved, but evaluating these pictures and diagrams is another skill. And depending on how well one knows their domain may or may not be able to make sense from them -- which again goes back to the [Research or business questions](#Research-or-business-questions).
+
+But try to visualise as much as possible at every opportunity available, to be able to understand the changing nature of the data to hand.
+
+[Back to the index](#index)
+
+### Sanity Check
+
+You always want to be sure that your data are _exactly_ how you want them to be, and because of this is a good rule of thumb to apply a sanity check after the completion of this step.
+
+Although sometimes we won't know "what sanity checks to apply?" till we have run through the rest of the steps in the process and then come back the next iteration(s) and fine-tune this step so that it catches the important low-hanging fruits that do not hinder the rest of the processes. Sometimes you might just have to proceed with the data and the steps to follow i.e. Data Preparation, Data Cleaning, Data Visualisation, etc... will help reveal much about your collected data and decisions to take.
```

Comment:
add some commas in the sentence, it improves readability.

Reply:
Yeah too many compound and complex sentences, I agree! I'll try to add commas or break them down!

Reply:
Done!